src/validation/transaction.cairo

```diff
@@ -28,6 +28,56 @@ pub fn validate_transaction(
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/consensus/tx_check.cpp
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/consensus/tx_verify.cpp
     //      - https://github.com/bitcoin/bitcoin/blob/master/src/validation.cpp
+    
+    if (*tx.inputs).len() == 0 {
+        return Result::Err("transaction inputs are empty");
+    };
+    if (*tx.outputs).len() == 0 {
+        return Result::Err("transaction outputs are empty");
+    };
+
+    if *tx.lock_time <= 0x1dcd64ff && *tx.lock_time > block_height {
```

Comment:
Please declare `LOCKTIME_THRESHOLD` as constant, and use decimal 500000000 value as it is used else where: https://github.com/bitcoin/bitcoin/blob/94c307b3c0746a08c6135ff38d6f1ad1ed6693bb/src/script/script.h#L46.